src/wallet/wallet2.cpp

```diff
@@ -2095,10 +2105,8 @@ void wallet2::process_new_transaction(const crypto::hash &txid, const cryptonote
         // the first one was already checked
         for (size_t i = 1; i < tx.vout.size(); ++i)
         {
-          tpool.submit(&waiter, boost::bind(&wallet2::check_acc_out_precomp_once, this, std::cref(tx.vout[i]), std::cref(derivation), std::cref(additional_derivations), i,
-            std::cref(is_out_data_ptr), std::ref(tx_scan_info[i]), std::ref(output_found[i])), true);
+          check_acc_out_precomp_once(tx.vout[i], derivation, additional_derivations, i, is_out_data_ptr, tx_scan_info[i], output_found[i]);
```

Comment:
I wonder whether we have to check for error in `tx_scan_info[i].error` after each call here like we do a few lines up for `tx_scan_info[0].error`. Maybe it's not necessary, maybe it's a slight omission already present in the old code.

Reply:
I can answer my own question: That gets checked only a few lines further down in code. Therefore no issue.